static/solo-cta/js/modal/menu.js

```diff
@@ -0,0 +1,255 @@
+            that.allVideos.push(newData[i]);
+            settings.data.push(newData[i]);
+            that.noVideosContainer.setAttribute('id', 'tvp-clearfix');
+            newData[i].duration = Utils.formatDuration(newData[i].duration);
+            that.noVideosContainer.innerHTML += Utils.tmpl(settings.templates['menu-item'], newData[i]);
+            that.scrollMenu.appendChild(that.noVideosContainer);
+            if (Utils.isset(settings, 'menu_item_play_category_tag_attribute') && ("" + settings.menu_item_play_category_tag_attribute).trim().length) {
+                var tagAttributeValue = newData[i][settings.menu_item_play_category_tag_attribute];
+                if (tagAttributeValue) {
+                    tagAttributeValue = tagAttributeValue.replace(/_/g,' ');
+                    var categoryFrag = document.createDocumentFragment(),
+                    categoryDiv = document.createElement('div');
+                    categoryDiv.classList.add('tvp-category-tag');
+                    categoryDiv.innerHTML += tagAttributeValue;
+                    categoryFrag.appendChild(categoryDiv);
+                    that.noVideosContainer.getElementsByClassName('tvp-video-details')[i].appendChild(categoryFrag);
+                }
+            }
+        }
+      }else{
+        var newVivFrag = document.createDocumentFragment(),
+            newDiv = document.createElement('div');
+        for (var i = 0; i < newData.length; i++) {
+            that.allVideos.push(newData[i]);
+            settings.data.push(newData[i]);
+            newDiv.setAttribute('id', 'tvp-clearfix');
+            newData[i].duration = Utils.formatDuration(newData[i].duration);
+            newDiv.innerHTML += Utils.tmpl(settings.templates['menu-item'], newData[i]);
+            newVivFrag.appendChild(newDiv);
+            
+            that.scrollMenu.appendChild(newDiv);
+            if (Utils.isset(settings, 'menu_item_play_category_tag_attribute') && ("" + settings.menu_item_play_category_tag_attribute).trim().length) {
+                var tagAttributeValue = newData[i][settings.menu_item_play_category_tag_attribute];
+                if (tagAttributeValue) {
+                    tagAttributeValue = tagAttributeValue.replace(/_/g,' ');
+                    var categoryFrag = document.createDocumentFragment(),
+                    categoryDiv = document.createElement('div');
+                    categoryDiv.classList.add('tvp-category-tag');
+                    categoryDiv.innerHTML += tagAttributeValue;
+                    categoryFrag.appendChild(categoryDiv);
+                    newDiv.getElementsByClassName('tvp-video-details')[i].appendChild(categoryFrag);
+                }
+            }
+        }
+      }
+      that.cacheDOM();
+      that.bindClickEvent();
+    };
+
+    this.deleteDivs = function(){
+        for (var i = that.tvpNoVideos.length - 1; i >= 0; i--){
+            that.noVideosContainer.removeChild(that.tvpNoVideos[i]);
```

Comment:
is there any events linked to these elements? and maybe there's a better way to flush the parent element